src/Appwrite/Platform/Tasks/ScheduleMessages.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Appwrite\Platform\Tasks;
+
+use Appwrite\Event\Delete;
+use Swoole\Timer;
+use Utopia\Platform\Action;
+use Utopia\CLI\Console;
+use Utopia\Database\DateTime;
+use Utopia\Database\Query;
+use Utopia\Database\Database;
+use Utopia\Pools\Group;
+use Appwrite\Event\Messaging;
+
+use function Swoole\Coroutine\run;
+
+class ScheduleMessages extends ScheduleBase
+{
+    public const UPDATE_TIMER = 3; // seconds
+    public const ENQUEUE_TIMER = 60; // seconds
+
+    public static function getName(): string
+    {
+        return 'schedule-messages';
+    }
+
+    public static function getSupportedResource(): string
+    {
+        return 'message';
+    }
+
+    protected function enqueueResources(Group $pools, Database $dbForConsole): void
+    {
+        foreach ($this->schedules as $schedule) {
+            \go(function () use ($schedule, $pools, $dbForConsole) {
```

Comment:
Granularity is per-minute not per-second like functions, since the timer ticks every minute, we don't need to check for messages that would send before the next tick

Reply:
Isn't functions also per-minute? 🧐 

I think the delayed execution was added to make sure the job is queued at the beginning of the minute. Let's say this worker started up at 2:00:50PM. Since the `enqueueResources()` method is called every minute:

https://github.com/appwrite/appwrite/blob/4d9fb1d6875131348657ca8c680a6a5dc89e089f/src/Appwrite/Platform/Tasks/ScheduleBase.php#L182-L185

, best-case, the job wouldn't be queued at 2:59:50PM and would have to wait until 3:00:50PM. This could be even worse if the schedules array is very large.

Reply:
We'll run into a coroutine limit before the size of the schedules array matters, I think the worst case will be your message gets scheduled 59 seconds after it's schedule time